org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java

```diff
@@ -0,0 +1,708 @@
+			.isAssignableFrom(throwableClass), "throwable of class",
+			fe -> fe.getThrowable()
+				.toString());
+
+		return allOf(throwableIsNotNull(), cond);//
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if a given type-mask <b>matches</b> the type.
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * static void example_type(int typeMask) {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(type(type))
+	 * 		.filteredOn(type(typeMask))
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(type(typeMask));// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param typeMask - the typeMask that would be checked against the bundle
+	 *            type of the {@link FrameworkEvent}
+	 * @return the Condition
+	 */
+	static Condition<FrameworkEvent> type(final int eventTypeMask) {
+
+		return VerboseCondition.verboseCondition(
+			(FrameworkEvent fe) -> Bitmap.typeMatchesMask(fe.getType(), eventTypeMask),
+			"type matches mask <" + FrameworkEventType.BITMAP.maskToString(eventTypeMask) + ">", //
+			fe -> " was <" + FrameworkEventType.BITMAP.toString(fe.getType()) + ">");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if a given type-mask <b>matches</b> the type <i>and</i> the
+	 * given {@link Bundle} <b>equals</b> the Bundle.
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * static void example_typeAndBundle(int typeMask, Bundle bundle) {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(typeAndBundle(typeMask, bundles))
+	 * 		.filteredOn(typeAndBundle(typeMask, bundle))
+	 * 		.first()// map to
+	 * 				// {@link
+	 * 				// ObjectAssert}
+	 * 		.has(typeAndBundle(typeMask, bundle));// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param typeMask - the typeMask that would be checked against the bundle
+	 *            type of the {@link FrameworkEvent}
+	 * @param bundle - the bundle that would be checked against the bundle of
+	 *            the {@link FrameworkEvent}
+	 * @return the Condition
+	 */
+	static Condition<FrameworkEvent> typeAndBundle(final int eventTypeMask, Bundle bundle) {
+		return Assertions.allOf(type(eventTypeMask), bundleEquals(bundle));
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if type of {@link FrameworkEvent} <b>matches</b>
+	 * <b>FrameworkEvent.ERROR</b>.
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * static void example_typeError() {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(typeError())
+	 * 		.filteredOn(typeError())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(typeError());// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<FrameworkEvent> typeError() {
```

Comment:
I don't think we want method for each event type.

Reply:
reduced to just one `type(int mask)` .